snapshots/devmapper/dmsetup/dmsetup.go

```diff
@@ -345,6 +353,39 @@ func BlockDeviceSize(path string) (int64, error) {
 	return size, nil
 }
 
+// DiscardBlocks discards all blocks for the given thin device
+//  ported from https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/pkg/devicemapper/devmapper.go#L416
+func DiscardBlocks(deviceName string) error {
+	inUse, err := isInUse(deviceName)
+	if err != nil {
+		return err
+	}
+	if inUse {
+		return ErrInUse
+	}
+	path := GetFullDevicePath(deviceName)
+	size, err := BlockDeviceSize(path)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(path, unix.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	request := [2]int64{
+		0,
+		size,
+	}
+	_, _, errno := unix.Syscall(unix.SYS_IOCTL, file.Fd(), C.BLKDISCARD, uintptr(unsafe.Pointer(&request[0])))
```

Comment:
We shouldn’t introduce cgo dependency for a single const

Reply:
The constant is architecture specific. It seems better to me to take the cgo dependency than to try to maintain the architecture specific values manually. 

Ideally this would be upstream in `golang.org/x/sys/unix`. Would you be comfortable taking the cgo dependency for now with a follow up to try to get it upstreamed?

Reply:
How about calling blkdiscard(8) instead of making a syscall? It would be slower, but more aligned with what the snapshotter does (using dmsetup instead of using the C library).

Reply:
That seems reasonable to me. I'll make that change.